cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetTest.java

```diff
@@ -116,6 +120,63 @@ public void testDecodeIncomplete() {
   }
 
   @Test
+  public void testPartitionMetadata() throws Exception {
```

Comment:
Were you thinking to wrap the different actions (read, scan, get, etc.) into separate transactions, or just one transaction for the test case?